crates/bevy_asset/src/loader.rs

```diff
@@ -359,10 +359,14 @@ impl<A: Asset> From<CompleteLoadedAsset<A>> for CompleteErasedLoadedAsset {
 /// [`NestedLoader::load`]: crate::NestedLoader::load
 /// [immediately]: crate::Immediate
 #[derive(Error, Debug)]
-#[error("Failed to load dependency {dependency:?} {error}")]
-pub struct LoadDirectError {
-    pub dependency: AssetPath<'static>,
-    pub error: AssetLoadError,
+pub enum LoadDirectError {
+    #[error("Requested to load an asset path ({0:?}) with a subasset, but this is unsupported")]
```

Comment:
I think this error message, despite trying its best, will be confusing to people that aren't pretty familiar with bevy's asset system. But maybe that's the target audience for now anyway? Maybe we should link one of the related issue(s)?

Reply:
Updated to link to a new issue.